### ISSUE TITLE
[Failing Test] `CP Migration Test Suite` is failing

### ISSUE THREAD

**Issue by ialidzhikov** [gardener/gardener#6703]
**How to categorize this issue?**

/area testing
/kind flake

**Which test(s)/suite(s) are flaking**:
`CP Migration Test Suite - /src/test/testmachinery/system/shoot_cp_migration`

**CI link**:
N/A

**Reason for failure**:
```
2022-09-19T16:49:10.382548576Z stdout F   In [AfterEach] at: /src/test/testmachinery/system/shoot_cp_migration/migrate_test.go:88
2022-09-19T16:49:10.38254363Z stdout F   The Shoot CP Migration health checks failed with: object: ClusterRole /extensions.gardener.cloud:provider-aws:aws-custom-route-controller Created At: 2022-09-19 16:43:41 +0000 UTC is created after the Shoot migration 2022-09-19 16:36:35.450222171 +0000 UTC m=+32.146907016
```

```
2022-09-18T15:19:34.003203851Z stdout F   In [AfterEach] at: /src/test/testmachinery/system/shoot_cp_migration/migrate_test.go:88
2022-09-18T15:19:34.00318197Z stdout F   The Shoot CP Migration health checks failed with: object: ClusterRole /extensions.gardener.cloud:provider-aws:aws-custom-route-controller Created At: 2022-09-18 15:13:34 +0000 UTC is created after the Shoot migration 2022-09-18 15:07:28.663129813 +0000 UTC m=+87.144110737
```

and many other failures with the same error.

**Anything else we need to know**:
N/A

**Comment by ialidzhikov** [comment]
cc @plkokanov @gardener/gardener-extension-provider-aws-maintainers

**Comment by plkokanov** [comment]
I had a look at this and the error is actually not due to control plane migration. 
There seems to be a problem with the custom route controller automatic enablement for k8s clusters > 1.22 that was added in https://github.com/gardener/gardener-extension-provider-aws/pull/612. What I noticed was that when a new cluster is created the `useCustomRouteController` is properly set to true by the webhook that mutates the `ControlPlane` resource. However, when the shoot is reconciled, the option is turned off.
Right after creation:
```
    providerConfig:
      apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
      cloudControllerManager:
        useCustomRouteController: true
```
After subsequent reconcilation:
```
    providerConfig:
      apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
```

The error was detected during the control plane migration tests, because the shoot clusters get reconciled before they get migrated as a toleration is added to them so they can be scheduled on protected seeds. During this reconciliation the option is turned off. Then the the shoot is migrated and the `ControlPlane` resource is newly created on the destination seed so the `useCustomRouteController` is automatically set to true as expected (I guess) and the `extensions.gardener.cloud:provider-aws:aws-custom-route-controller` is recreated.

@dkistner @kon-angelo @ScheererJ @rfranzke

**Comment by plkokanov** [comment]
@ialidzhikov should we close this now, since we know what the issues is and will come to a solution that also does not interfere with CP migration in https://github.com/gardener/gardener-extension-provider-aws/issues/619

**Comment by ialidzhikov** [comment]
Yes, let's close as the issue should be fixed with [`provider-aws@v1.38.2`](https://github.com/gardener/gardener-extension-provider-aws/releases/tag/v1.38.2).
Thank you for discovering this issue and participating in the discussion on how to tackle it! 🙇 It wasn't an easy one to discover. 

/close
